### ISSUE TITLE
Why is it slow to segment everything？ Is there a good solution？

### ISSUE THREAD

**Issue by shuaifengj** [yformer/EfficientSAM#69]

**Comment by silinsi** [comment]
Yes, it's slow and consumes too much memory, with 20GB memory it even can't support GRID_SIZE=32.

With GRID_SIZE=16 , image size=1920*1084, on RTX3090 , it costs 25seconds , much slower than other SAM models.

**Comment by CurryKd7** [comment]
> 是的，它很慢并且消耗太多内存，20GB 内存甚至无法支持 GRID_SIZE=32。
> 
> 当 GRID_SIZE=16 时，图像大小=1920*1084，在 RTX3090 上，它花费 25 秒，比其他 SAM 模型慢得多。

Have you solved the problem of slow segmentation?

**Comment by silinsi** [comment]
> > 是的，它很慢并且消耗太多内存，20GB 内存甚至无法支持 GRID_SIZE=32。
> > 当 GRID_SIZE=16 时，图像大小=1920*1084，在 RTX3090 上，它花费 25 秒，比其他 SAM 模型慢得多。
> 
> Have you solved the problem of slow segmentation?

Nope, in fact I just turned to EfficientViT-SAM to finish my graduation design months ago.

**Comment by CurryKd7** [comment]
> > > 是的，它很慢并且消耗太多内存，20GB 内存甚至无法支持 GRID_SIZE=32。
> > > 当 GRID_SIZE=16 时，图像大小=1920*1084，在 RTX3090 上，它花费 25 秒，比其他 SAM 模型慢得多。
> > 
> > 
> > 你解决了分割慢的问题吗？
> 
> 不，事实上我几个月前刚刚转向 EfficientViT-SAM 来完成我的毕业设计。

May I ask how efficient the iT SAM segmentation is and if there has been any improvement in speed?

**Comment by silinsi** [comment]
> > > > 是的，它很慢并且消耗太多内存，20GB 内存甚至无法支持 GRID_SIZE=32。
> > > > 当 GRID_SIZE=16 时，图像大小=1920*1084，在 RTX3090 上，它花费 25 秒，比其他 SAM 模型慢得多。
> > > 
> > > 
> > > 你解决了分割慢的问题吗？
> > 
> > 
> > 不，事实上我几个月前刚刚转向 EfficientViT-SAM 来完成我的毕业设计。
> 
> May I ask how efficient the iT SAM segmentation is and if there has been any improvement in speed?

It takes about twice as much time as Mobile SAM does.(image size 1920*1080, grid size 32 each side, seg-every takes about 7.6s on single RTX3090 ). 
About the improvement, my recent use of it is in May this year so you should check it out yourself.

**Comment by CurryKd7** [comment]
> > > > > 是的，它很慢并且消耗太多内存，20GB 内存甚至无法支持 GRID_SIZE=32。
> > > > > 当 GRID_SIZE=16 时，图像大小=1920*1084，在 RTX3090 上，它花费 25 秒，比其他 SAM 模型慢得多。
> > > > 
> > > > 
> > > > 你解决了分割慢的问题吗？
> > > 
> > > 
> > > 不，事实上我几个月前刚刚转向 EfficientViT-SAM 来完成我的毕业设计。
> > 
> > 
> > 请问 iT SAM 分段的效率如何，速度是否有任何提高？
> 
> 它花费的时间大约是 Mobile SAM 的两倍。（图像尺寸 1920*1080，网格大小每侧 32，单RTX3090每个段大约需要 7.6 秒）。关于改进，我最近使用它是在今年 5 月，所以你应该自己检查一下。

Why did I run EfficientViT SAM code to segment a single image without much improvement in time? May I ask if youmodel: EfficientViTSam,points_per_side: int or None = 32,points_per_batch: int = 64,pred_iou_thresh: float = 0.88,stability_score_thresh: float = 0.95,stability_score_offset: float = 1.0,box_nms_thresh: float = 0.7,crop_n_layers: int = 1,crop_nms_thresh: float = 0.7,crop_overlap_ratio: float = 512 / 1500,crop_n_points_downscale_factor: int = 1,point_grids: list[np.ndarray] or None = None,min_mask_region_area: int = 25,How are these parameters set for outputting mode: str="binary_mask"

**Comment by yformer** [comment]
Hi @shuaifengj @silinsi @CurryKd7, thanks for your interest in our work!

I think the current segment everything code did not optimize well.

We have updated EfficientSAM for unified image and video segmentation. Can you give it a try?

Please check https://github.com/yformer/EfficientTAM and segment everything demo, https://5239f8e221db7ee8a0.gradio.live/
